### PR TITLE
Bump LLVM to 18.1.8

### PR DESCRIPTION
**Pull request by singiamtel**
**Do not merge**. Only testing CI for now

Testing whether LLVM 18 works.

Corresponding LLVM tag https://github.com/alisw/llvm-project-reduced/tree/alice/llvmorg-18.1.8

### CODE REVIEW

File: clang.sh [hunk 1]
```diff
@@ -49,14 +49,15 @@ cmake "$SOURCEDIR/llvm" \
...
+  -DLIBCXXABI_USE_LLVM_UNWINDER=OFF \
   ${GCC_TOOLCHAIN_ROOT:+-DGCC_INSTALL_PREFIX=$GCC_TOOLCHAIN_ROOT}
 
 cmake --build . -- ${JOBS:+-j$JOBS} install
 
 if [[ $a == v18.1.* ]]; then
   SPIRV_TRANSLATOR_VERSION="v18.1.3"
 else
-  SPIRV_TRANSLATOR_VERSION="${PKGVERSION%%.*}.0.0"
+  SPIRV_TRANSLATOR_VERSION="${PKGVERSION%.*}.0"
```
[the remark this answers is not in the document]
Comment (davidrohr) [reply clang.sh:60]: The above check should be `$PKGVERSION == v18.1.*` instead of `a ==`.
You should use 18.1 for LLVM SPIRV translator

**Comment by davidrohr** [comment]
And please check with @aalkin : There is some problem with LLVM 18 and apache arrow for his use case, that is why we bumped to 17 only.

**Comment by aalkin** [comment]
Yes, indeed, arrow will compile with LLVM 18, but in the runtime it will crash our workflows exclusively on aarch64.

**Comment by aalkin** [comment]
The mac-aarch64 issue should be solved by #5579. Mac testers still use older llvm currently.

**Comment by ktf** [comment]
O2CodeChecker needs to be adapted accordingly...

**Comment by davidrohr** [comment]
@singiamtel : Do you know how to update O2CodeChecker?
You can compare to what I did for LLVM 17: https://github.com/AliceO2Group/O2CodeChecker/commit/1f7dc38543c634d0ad9851e992326dde87a9ce3f
One just has to copy over the relevant header files from this folder: https://github.com/llvm/llvm-project/tree/main/clang-tools-extra/clang-tidy
Sometimes, when the API changes, one also has to adjust tool/ClangTidyMain.cpp (like here https://github.com/AliceO2Group/O2CodeChecker/commit/9b46b24114792e4b18405371292ad3d2582969f5#diff-ec57b2e744cc0a1055ac2a5cbe1a60042ec69bed0dfcebe16fed83a22b543431) analogous to the changes in the new LLVM clang-tidy tool.

**Comment by davidrohr** [comment]
I believe we need the attached patch for Clang to fix the
```
InvalidInstruction: Can't translate llvm instruction:
 Global variable cannot have Function storage class. Consider setting a proper address space.
 Original LLVM value:
 ```
[spirv.patch.gz](https://github.com/user-attachments/files/16710434/spirv.patch.gz)

**Comment by davidrohr** [comment]
ALternatively, this is the latest version of the fix: https://github.com/llvm/llvm-project/pull/90048/files

**Comment by davidrohr** [comment]
My clang 18 works since gentoo pulled in that patch

**Review by davidrohr — APPROVED** [review]
yeah, finally this looks good! LGTM!

**Comment by ktf** [comment]
merging so that we get it in tomorrow build.

**Pull request merged**